cmake/onnxruntime_providers.cmake

```diff
@@ -669,6 +669,10 @@ if (onnxruntime_USE_COREML)
     message(FATAL_ERROR "CoreML EP can not be used in a basic minimal build. Please build with '--minimal_build extended'")
   endif()
 
+  if(NOT APPLE)
+    message(FATAL_ERROR "CoreML EP must be built on an Apple platform.")
+  endif()
+
   add_compile_definitions(USE_COREML=1)
```

Comment:
Not sure this is correct. We need to support a build on non-apple platforms in order to create an ORT format model that's aware of what the CoreML EP can handle. 

This build only supports GetCapability, and the Compile returns an error (e.g. [https://github.com/microsoft/onnxruntime/blob/cdd2129fed4c7583011338485143e641084366a0/onnxruntime/core/providers/nnapi/nnapi_builtin/nnapi_execution_provider.cc#L459-L474](https://github.com/microsoft/onnxruntime/blob/cdd2129fed4c7583011338485143e641084366a0/onnxruntime/core/providers/nnapi/nnapi_builtin/nnapi_execution_provider.cc#L459-L474))

Reply:
At least until we implement runtime replay of optimizations, as we wouldn't need to create an ORT format model that's aware of what nodes NNAPI or CoreML could take.

Reply:
The CoreML EP has some Objective-C++ files. Does that mean we support building those on non-Apple platforms?

Reply:
The GetCapability of CoreML EP is not using any objc code or any apple platform library.
But for now the CoreML EP, GetCapability and Compile are not yet separated, so this change is ok. 
We can separate those later when we add coreml to the conversion script. (need to add a bunch of #ifdef and change the cmake files)

Reply:
handling this another way now, removed the if(APPLE) check